app/models/case/sar/standard.rb

```diff
@@ -146,11 +151,20 @@ def reset_deadline!
     )
   end
 
-  # SARs extensions are based on calendar days not working days
+  # The deadlines are all calculated based on the date receiving the case
   def max_allowed_deadline_date
-    initial_deadline + Settings.sar_extension_limit.to_i.days
+    time_limit = self.correspondence_type.extension_time_limit || Settings.sar_extension_default_limit
+    deadline_calculator.max_allowed_deadline_date(time_limit)
   end
 
+  def extension_time_limit
+    self.correspondence_type.extension_time_limit || Settings.sar_extension_default_limit
+  end 
+
+  def extension_time_default
+    self.correspondence_type.extension_time_default || Settings.sar_extension_default_time_default
+  end 
+
```

Comment:
Probably me, but I'm really struggling with the permutations of extension, time, default here :) potentially a bit confusing?
Is there any way we can make the purpose of these settings (and the difference between them) a bit clearer?

I think `sar_extension_default_time_default` has jumped the shark - anything that needs the word "default" in it twice needs to be split up into at least two other things, I think :)

Reply:
hmm, it must be my bad english sense,  let me rename it to "sar_extension_default_gap", it means the time-gap when you extent it every tme